core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -531,6 +528,9 @@ public void delete(Engine.Delete delete) {
             throw ex;
         }
         indexingService.postDelete(delete);
```

Comment:
if we remove the removeWritingBytes / addWritingBytes and just use a `AtomicLong` on IndexShard we can move  `indexingMemoryController.bytesWritten` into `indexingService.postDelete` / `indexingService.postIndex` and therefore can remove the dependency between IndexShard and IndexMemoryController which I think we should not have, What do you think @mikemccand